store/src/main/scala/geotrellis/store/util/BlockingThreadPool.scala

```diff
@@ -14,12 +14,14 @@
  * limitations under the License.
  */
 
-package geotrellis.util
+package geotrellis.store.util
 
-import pureconfig.generic.ProductHint
-import pureconfig.{CamelCase, ConfigFieldMapping}
+import geotrellis.store.conf.BlockingThreadPoolConfig
 
+import java.util.concurrent.{ExecutorService, Executors}
+import scala.concurrent.ExecutionContext
 
-trait CamelCaseConfig {
-  implicit def hint[T]: ProductHint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
+object BlockingThreadPool extends Serializable {
+  @transient lazy val pool: ExecutorService = Executors.newFixedThreadPool(BlockingThreadPoolConfig.conf.threads)
+  @transient lazy val executionContext: ExecutionContext = ExecutionContext.fromExecutor(pool)
```

Comment:
This may be a comment to be satisfied by another PR/issue, but I think it's useful to ask: How is this thread pool going to be named to facilitate debugging? When a user connects to VM is it going to be `BlockingThreadPool` and what will be the names of threads? It would be ideal if they had names like `geotrellis-defualt-io-thread-##`  and belong to thread pool that somehow was connected to GeoTrellis project.

Reply:
@echeipesh `geotrellis-default-io-%d` now